src/connectors/sort-by-selector/connectSortBySelector.js

```diff
@@ -157,6 +157,30 @@ export default function connectSortBySelector(renderFn, unmountFn) {
         unmountFn();
         return state.setIndex(this.initialIndex);
       },
+
+      getWidgetState(fullState, { state }) {
+        const currentIndex = state.getQueryParameter('index');
+        const isInitialIndex = currentIndex === this.initialIndex;
+
+        if (
+          isInitialIndex ||
+          (fullState && fullState.sortBy === currentIndex)
```

Comment:
As said in the slack thread, there's currently an issue on how we compute the default value for the index that we consider when we want to remove the index from the url.

Reply:
The initial index is now computed from the parameter `indexName` given to `instantsearch`